job-machine/src/components/auth/FormCommon/FormContainer.tsx

```diff
@@ -0,0 +1,126 @@
+import { useEffect, useState } from 'react';
+import { useForm, Controller } from 'react-hook-form';
+import { Typography, Form, Input } from 'antd';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import FormIcon from './FormIcon';
+import { ContainerForm } from './FormContainer.styled';
+import { BaseButton } from 'components/common/BaseButton/BaseButton';
+import {
+  Data,
+  FormLoginType,
+  FormRegisterType,
+  TypeActivePanel,
+} from 'interfaces/interfaces';
+import { EMAIL_REGEX, dataSignIn, dataSignUp } from 'constants/constants';
+import { WrapperFormItem } from './WrapperFormItem.styled';
+import { useDispatch } from 'react-redux';
+import {
+  loginRequest,
+  registerRequest,
+} from '../../../redux/actions/authAction';
+
+const { Title } = Typography;
+
+type FormContainerProps = {
+  state: TypeActivePanel;
+};
+
+const FormContainer = ({ state }: FormContainerProps) => {
+  const [formData, setFormData] = useState<Data>(dataSignIn);
+  const checkSignIn = state === 'sign-in';
+  const schema = yup.object().shape({
+    name: checkSignIn ? yup.string() : yup.string().min(8).max(32).required(),
+    email: yup
+      .string()
+      .required('Email is required')
+      .matches(EMAIL_REGEX, 'Email invalid '),
+
+    password: yup.string().min(8).max(32).required(),
+  });
+
+  const dispatch = useDispatch();
+  const {
+    handleSubmit,
+    control,
+    reset,
+    formState: { errors },
+  } = useForm<FormLoginType | FormRegisterType>({
+    defaultValues: {
+      name: '',
+      email: '',
+      password: '',
+    },
+    resolver: yupResolver<FormLoginType | FormRegisterType>(schema),
+  });
+  let timeoutId: NodeJS.Timeout;
+
+  function returnData(): Promise<Data> {
+    return new Promise(resolve => {
+      timeoutId = setTimeout(() => {
+        const data = state === 'sign-in' ? dataSignIn : dataSignUp;
+        resolve(data);
```

Comment:
`resolve( state === 'sign-in' ? dataSignIn : dataSignUp;)` có thể viết thẳng vào như này